tidb connection running loop panic happened while chaos testing

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

- add workload to tidb.
- inject chaos(az network partition)

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)
No error happened

### 3. What did you see instead (Required)
tidb panic
[2022/12/05 17:01:52.276 +08:00] [ERROR] [conn.go:1058] ["connection running loop panic"] [conn=468473317293034745] [lastSQL="select   [runs.id](http://runs.id/),\n                    runs.jobId,\n                    runs.triggerId,\n                    runs.originalAttemptId,\n                    runs.idInJob,\n                    runs.userId,\n                    runs.orgId,\n                    runs.lifeCycleState,\n                    runs.lifeCycleSubState,\n                    runs.clusterId,\n                    runs.sparkContextId,\n                    runs.scheduledTime,\n                    runs.creationTime,\n                    runs.terminationTime,\n                    runMetadata.rootRunId,\n                    runMetadata.parentRunId,\n                    runMetadata.jobType,\n                    runMetadata.useRunnerService,\n                    runMetadata.runnerId,\n                    runMetadata.sessionId,\n                    runMetadata.numRunnerAssignments,\n                    runs.multiTaskParentRunId as parentJobRunId,\n                    runs.taskId,\n                    runs.terminationReason,\n                    runs.lifeCycleSubState,\n                    runs.lifeCycleState,\n                    runs.runCancellationReason,\n                    runs.userVisibleState,\n                    runs.clusterWaitSec,\n                    runs.runSec,\n                    runs.cleanupSec,\n                    runs.clusterTerminateSec,\n                    runs.dataVersion,\n                    runs.version,\n                    runs.taskKey,\n                    runs.triggerInfo,\n                    runs.repoMetadataSnapshot,\n                    runs.dependencyJson,\n                    runs.description,\n                    runMetadata.taskStates,\n                    runMetadata.extraContext,\n                    runMetadata.idempotentToken,\n\n                    runMetadata.runName,\n                    runMetadata.clusterJson,\n                    runs.miscMessage\n                    from\n             runs left join runMetadata on [runs.id](http://runs.id/) = [runMetadata.id](http://runmetadata.id/)\n             and runs.orgId = runMetadata.runMetadataOrgId\n           where\n             runs.orgId = ?\n             and [runs.id](http://runs.id/) = ? [arguments: (2943190370600165, 89038380001)]"] [err="interface conversion: executor.Executor is nil, not *executor.IndexLookUpJoin"] [stack="[github.com/pingcap/tidb/server.(*clientConn).Run.func1\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:1061\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:884\ngithub.com/pingcap/tidb/executor.(*ExecStmt).Exec.func1\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/adapter.go:440\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:884\nruntime.panicdottypeE\n\t/usr/local/go/src/runtime/iface.go:262\nruntime.panicdottypeI\n\t/usr/local/go/src/runtime/iface.go:272\ngithub.com/pingcap/tidb/executor.(*executorBuilder).buildIndexNestedLoopHashJoin\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/builder.go:3288\ngithub.com/pingcap/tidb/executor.(*executorBuilder).build\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/builder.go:261\ngithub.com/pingcap/tidb/executor.(*executorBuilder).buildProjection\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/builder.go:1685\ngithub.com/pingcap/tidb/executor.(*executorBuilder).build\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/builder.go:269\ngithub.com/pingcap/tidb/executor.(*ExecuteExec).Build\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/prepared.go:178\ngithub.com/pingcap/tidb/executor.(*ExecStmt).buildExecutor\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/adapter.go:1110\ngithub.com/pingcap/tidb/executor.(*ExecStmt).Exec\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/adapter.go:500\ngithub.com/pingcap/tidb/session.runStmt\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/session/session.go:2329\ngithub.com/pingcap/tidb/session.(*session).ExecuteStmt\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/session/session.go:2193\ngithub.com/pingcap/tidb/server.(*TiDBContext).ExecuteStmt\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/driver_tidb.go:233\ngithub.com/pingcap/tidb/server.(*clientConn).executePreparedStmtAndWriteResult\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn_stmt.go:262\ngithub.com/pingcap/tidb/server.(*clientConn).executePlanCacheStmt\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn_stmt.go:221\ngithub.com/pingcap/tidb/server.(*clientConn).handleStmtExecute\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn_stmt.go:213\ngithub.com/pingcap/tidb/server.(*clientConn).dispatch\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:1401\ngithub.com/pingcap/tidb/server.(*clientConn).Run\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:1123\ngithub.com/pingcap/tidb/server.(*Server).onConn\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/server.go:624](http://github.com/pingcap/tidb/server.(*clientConn).Run.func1/n/t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:1061/nruntime.gopanic/n/t/usr/local/go/src/runtime/panic.go:884/ngithub.com/pingcap/tidb/executor.(*ExecStmt).Exec.func1/n/t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/adapter.go:440/nruntime.gopanic/n/t/usr/local/go/src/runtime/panic.go:884/nruntime.panicdottypeE/n/t/usr/local/go/src/runtime/iface.go:262/nruntime.panicdottypeI/n/t/usr/local/go/src/runtime/iface.go:272/ngithub.com/pingcap/tidb/executor.(*executorBuilder).buildIndexNestedLoopHashJoin/n/t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/builder.go:3288/ngithub.com/pingcap/tidb/executor.(*executorBuilder).build/n/t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/builder.go:261/ngithub.com/pingcap/tidb/executor.(*executorBuilder).buildProjection/n/t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/builder.go:1685/ngithub.com/pingcap/tidb/executor.(*executorBuilder).build/n/t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/builder.go:269/ngithub.com/pingcap/tidb/executor.(*ExecuteExec).Build/n/t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/prepared.go:178/ngithub.com/pingcap/tidb/executor.(*ExecStmt).buildExecutor/n/t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/adapter.go:1110/ngithub.com/pingcap/tidb/executor.(*ExecStmt).Exec/n/t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/adapter.go:500/ngithub.com/pingcap/tidb/session.runStmt/n/t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/session/session.go:2329/ngithub.com/pingcap/tidb/session.(*session).ExecuteStmt/n/t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/session/session.go:2193/ngithub.com/pingcap/tidb/server.(*TiDBContext).ExecuteStmt/n/t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/driver_tidb.go:233/ngithub.com/pingcap/tidb/server.(*clientConn).executePreparedStmtAndWriteResult/n/t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn_stmt.go:262/ngithub.com/pingcap/tidb/server.(*clientConn).executePlanCacheStmt/n/t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn_stmt.go:221/ngithub.com/pingcap/tidb/server.(*clientConn).handleStmtExecute/n/t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn_stmt.go:213/ngithub.com/pingcap/tidb/server.(*clientConn).dispatch/n/t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:1401/ngithub.com/pingcap/tidb/server.(*clientConn).Run/n/t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:1123/ngithub.com/pingcap/tidb/server.(*Server).onConn/n/t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/server.go:624)"]
### 4. What is your TiDB version? (Required)
# /tidb-server -V
Release Version: v6.5.0
Edition: Community
Git Commit Hash: f799e720f5aec32664c6910e47bad46e5c838b4f
Git Branch: heads/refs/tags/v6.5.0
UTC Build Time: 2022-12-02 08:57:01
GoVersion: go1.19.3
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: unistore
<!-- Paste the output of SELECT tidb_version() -->

--- a/executor/builder.go
+++ b/executor/builder.go
@@ -3285,7 +3285,11 @@ func (b *executorBuilder) buildIndexLookUpMergeJoin(v *plannercore.PhysicalIndex
 }
 
 func (b *executorBuilder) buildIndexNestedLoopHashJoin(v *plannercore.PhysicalIndexHashJoin) Executor {
-	e := b.buildIndexLookUpJoin(&(v.PhysicalIndexJoin)).(*IndexLookUpJoin)
+	join := b.buildIndexLookUpJoin(&(v.PhysicalIndexJoin))
+	if b.err != nil {
+		return nil
+	}
+	e := join.(*IndexLookUpJoin)
 	idxHash := &IndexNestedLoopHashJoin{
 		IndexLookUpJoin: *e,
 		keepOuterOrder:  v.KeepOuterOrder,

